awx/main/credential_plugins/centrify_vault.py

```diff
@@ -0,0 +1,135 @@
+from .plugin import CredentialPlugin, raise_for_status
+from django.utils.translation import ugettext_lazy as _
+from urllib.parse import urljoin
+import requests
+import base64
+import json
+pas_inputs = {
+    'fields': [{
+        'id': 'url',
+        'label': _('Centrify Tenant URL'),
+        'type': 'string',
+        'help_text': _('Centrify Tenant URL'),
+        'format': 'url',
+    }, {
+        'id':'client_id',
+        'label':_('Centrify API User'),
+        'type':'string',
+        'help_text': _('Centrify API User, having necessary permissions as mentioned in support doc'),
+
+    }, {
+        'id':'client_password',
+        'label':_('Centrify API Password'),
+        'type':'string',
+        'help_text': _('Password of Centrify API User with necessary permissions'),
+        'secret':True,
+    }],
+    'metadata': [{
+        'id': 'account-name',
+        'label': _('Account Name'),
+        'type': 'string',
+        'help_text': _('Local system account or Domain account name enrolled in Centrify Vault. eg. (root or DOMAIN/Administrator)'),
+    },{
+        'id': 'system-name',
+        'label': _('System Name'),
+        'type': 'string',
+        'help_text': _('Machine Name enrolled with in Centrify Portal'),
+    }],
+    'required': ['url', 'account-name', 'system-name','client_id','client_password'],
+}
+
+
+# generate bearer token to authenticate with PAS portal, Input : Client ID, Client Secret
+def handle_auth(**kwargs):
+    tokens = None
+    post_data = {
+        "grant_type": "client_credentials", 
+        "scope":"siem" 
+    }
+    post_header = {
+        "Authorization": 'Basic ' + base64.b64encode(bytes(kwargs['client_id'] + ":" + kwargs['client_password'], 'ascii')).decode('ascii')
+    }
+    response = requests.post(
+        kwargs['endpoint'],
+        data = post_data,
+        headers = post_header, 
+        verify = True,
+        timeout = (5, 30)
+    )
+    raise_for_status(response)
+    tokens = json.loads(response.text)
+    try: 
+        return tokens['access_token']
+    except KeyError:
+        raise RuntimeError('OAuth request to tenant was unsuccessful')
+
+
+# fetch the ID of system with RedRock query, Input : System Name, Account Name
+def get_ID(**kwargs):
+    endpoint = urljoin(kwargs['url'],'/Redrock/query')
+    name=" Name='{0}' and User='{1}'".format(kwargs['system_name'],kwargs['acc_name'])

```

Comment:
I'm not intimately familiar with Centrify Vault and how it works, but is this a SQL query being passed as an HTTP argument?  Is this input sanitized in some way on the backend?  If not, this code seems like it would open to a SQL injection attack by somebody who crafted specific `system_name` and `acc_name` values.

Reply:
Hey @surbhijain1502 @Asharma-bhavna,

Thanks for the updated commits addressing by feedback.  Do you all have any additional context you can provide about this comment?

Thanks!

Reply:
@ryanpetrello 
O Yes ryan, My bad!! I forgot of this.  We discussed this internally. It's completely safe to use this query.  So here this thing, this endpoint doesn't allow any write request explicitly or if embedded in where clause as well to execute. Moreover, every action is permission controlled, so making read also secure.

Reply:
@ryanpetrello  Hope, your doubts been resolved. Also, could you help us in identifying why zuul build automation process failed after our latest changes? How we can debug it?

Reply:
@surbhijain1502 @Asharma-bhavna,

I've installed a Centrify Vault account and I'm trying to work through the actual intended workflow here so that I can test this PR and verify that it works.  Can you explain how these parameters (`account-name` and `system-name`) map to Resources I would create in Centrify free trial?  The more documented steps and descriptions that you can provide explaining the workflow of how you tested this PR, the better.

Reply:
Specifically, I'm looking for the minimal set of steps to follow to arrive at this screen in AWX to successfully test the connectivity of this plugin:

<img width="1181" alt="image" src="https://user-images.githubusercontent.com/214912/109718323-2b227280-7b75-11eb-91d1-b28ac35e0964.png">

Reply:
This is as far as I've gotten in my testing using a trial Centrify Vault I installed via https://www.centrify.com/trial-center/

```python
>>> response.status_code
400
>>> response.text
'{"error":"invalid_request","error_description":"unknown app oauthsiem"}'
```